engine/src/main/java/io/zeebe/engine/state/appliers/ProcessInstanceElementActivatingApplier.java

```diff
@@ -30,6 +31,17 @@ public void applyState(final long elementInstanceKey, final ProcessInstanceRecor
     elementInstanceState.newInstance(
         flowScopeInstance, elementInstanceKey, value, ProcessInstanceIntent.ELEMENT_ACTIVATING);
 
+    final var flowScopeElementType = flowScopeInstance.getValue().getBpmnElementType();
+    final var currentElementType = value.getBpmnElementType();
+    if ((currentElementType == BpmnElementType.START_EVENT
+            && (flowScopeElementType == BpmnElementType.SUB_PROCESS
+                || flowScopeElementType == BpmnElementType.PROCESS))
+        || flowScopeElementType == BpmnElementType.MULTI_INSTANCE_BODY) {
+      // we currently spawn new tokens only for container elements
+      // we might spawn tokens for other bpmn element types as well later, then we can improve here
+      elementInstanceState.spawnToken(flowScopeInstance.getKey());
```

Comment:
It isn't clear to me why we spawn a token here, but it's been a while since I worked in the engine so I'm sure it's just me missing something. Can you explain why? Isn't it enough to create the child element instance? 

We can also do that via Zoom :+1:

Reply:
https://github.com/camunda-cloud/zeebe/pull/6524#discussion_r592162941